src/subcommands/add.go

```diff
@@ -93,93 +93,105 @@ func Add(a AddArgs) {
-		err := decoder.Decode(&entry)
-		if err == io.EOF {
-			// Reached end of file, entry not found
-			break
-		}
-		if err != nil {
-			fmt.Printf("Failed to decode entry: %s\n", err)
-			os.Exit(1)
-		}
-		if entry.Name == ale.Name {
-			return true
+	var index pb.Index
+	if err := proto.Unmarshal(data, &index); err != nil {
+		return false, fmt.Errorf("failed to marshall index file: %w", err)
+	}
+
+	for _, entry := range index.Entries {
+		if entry.Name == name {
+			return true, nil
 		}
 	}
-	return false
+
+	// If the name is not found in the index file, return false
+	return false, nil
 }
 
-func commitAdd(ale AddedListEntry, fp *os.File) error {
-	// Seek to the end of the file
-	_, err := fp.Seek(0, io.SeekEnd)
-	if err != nil {
-		return fmt.Errorf("failed to seek to end of file: %w", err)
+// commitAdd appends the provided IndexListEntry to the index file specified by indexFilePath.
+// It reads the existing index file data, unmarshals it, appends the provided entry, marshals the updated index data,
+// and writes it back to the index file. It returns an error if any operation fails.
+//
+// Parameters:
+//
+//	ile *pb.IndexListEntry: The IndexListEntry to be appended to the index entries.
+//	indexFilePath string: The file path of the index file.
+func commitAdd(ile *pb.IndexListEntry, indexFilePath string) error {
+
+	var index pb.Index
+	data, err := os.ReadFile(indexFilePath)
+
+	if err == nil {
+		if err := proto.Unmarshal(data, &index); err != nil {
+			return fmt.Errorf("failed to unmarshal existing index: %w", err)
+		}
+	} else if !os.IsNotExist(err) {
+		return fmt.Errorf("failed to read index file: %w", err)
 	}
 
-	// Create an encoder that will append to the file
-	encoder := gob.NewEncoder(fp)
+	index.Entries = append(index.Entries, ile)
 
-	// Encode and append the new entry
-	err = encoder.Encode(&ale)
+	updatedData, err := proto.Marshal(&index)
 	if err != nil {
-		return fmt.Errorf("failed to encode and append new entry: %w", err)
+		return fmt.Errorf("failed to marshal updated index: %w", err)
+	}
+	if err := os.WriteFile(indexFilePath, updatedData, 0644); err != nil {
+		return fmt.Errorf("failed to write updated index to file: %w", err)
 	}
 
 	return nil
 }
 
-func copyFile(src string, dest string, ale AddedListEntry) error {
-	// Open the source file
+// copyFile copies the content of the source file to the destination file and updates the given IndexListEntry with the content and last refreshed time.
+// Parameters:
+//
+//	src: the path of the source file to be copied
+//	dest: the path of the destination file where the content will be copied to
+//	ile: a pointer to an IndexListEntry to be written to the destination file
+func copyFile(src string, dest string, ile *pb.IndexListEntry) error {
 	srcFile, err := os.Open(src)
 	if err != nil {
 		return fmt.Errorf("failed to open source file: %w", err)
 	}
 	defer srcFile.Close()
 
-	// Read the entire file content
 	content, err := io.ReadAll(srcFile)
```

Comment:
I don't get it, won't this save the entire copied file contents (including other fields) into the new file content field?

Reply:
If this is a bug/oversight as serious as I think it is, you should probably add or update a unit test to verify that copied files get their contents properly copied

Reply:
Yes it will copy the indexListEntry of the file including the contents. It will make a copy of the current file including the contents and other fields

Reply:
Oops, I am being stupid. We do want to copy the content. For some reason I thought we were copying an index lol